### ISSUE TITLE
Update of stoken to stoken-0.92-r2.apk in testing brakes docker build

### ISSUE THREAD

**Issue by AndriyKarpyak** [alpinelinux/docker-alpine#90]
Hello team!

I'm getting the next error:

```
fetch http://dl-3.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  so:libhogweed.so.6 (missing):
    required by: stoken-0.92-r2[so:libhogweed.so.6]
  so:libnettle.so.8 (missing):
    required by: stoken-0.92-r2[so:libnettle.so.8]
```

during the build of the image.

Problem is triggered in response to the installation of openconnect

`apk add openconnect --no-cache --repository http://dl-3.alpinelinux.org/alpine/edge/testing/ --allow-untrusted`

This morning build passed without problem but from what I see in logs difference is in stoken versions. 
```
2020-06-02T10:59:13.1498320Z fetch http://dl-3.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
2020-06-02T10:59:13.4677330Z fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
2020-06-02T10:59:13.5285213Z fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
2020-06-02T10:59:13.6807619Z (1/24) Installing libgpg-error (1.36-r2)
2020-06-02T10:59:13.6907785Z (2/24) Installing libgcrypt (1.8.5-r0)
2020-06-02T10:59:13.7149136Z (3/24) Installing vpnc (0.5.3-r8)
2020-06-02T10:59:14.0170068Z (4/24) Installing libmnl (1.0.4-r0)
2020-06-02T10:59:14.0252784Z (5/24) Installing libnftnl-libs (1.1.5-r0)
2020-06-02T10:59:14.0350891Z (6/24) Installing iptables (1.8.3-r2)
2020-06-02T10:59:14.0804122Z (7/24) Installing iproute2 (5.4.0-r1)
2020-06-02T10:59:14.1135727Z Executing iproute2-5.4.0-r1.post-install
2020-06-02T10:59:14.1136658Z (8/24) Installing lz4-libs (1.9.2-r0)
2020-06-02T10:59:14.1236040Z (9/24) Installing pcsc-lite-libs (1.8.25-r2)
2020-06-02T10:59:14.1325161Z (10/24) Installing libproxy (0.4.15-r4)
2020-06-02T10:59:14.1440794Z (11/24) Installing perl (5.30.1-r0)
2020-06-02T10:59:14.5184244Z (12/24) Installing perl-error (0.17028-r0)
2020-06-02T10:59:14.5259973Z (13/24) Installing perl-git (2.24.3-r0)
2020-06-02T10:59:14.5355984Z (14/24) Installing git-perl (2.24.3-r0)
2020-06-02T10:59:14.5439284Z (15/24) Installing cvs (1.12.12-r0)
2020-06-02T10:59:14.5761357Z (16/24) Installing diffutils (3.7-r0)
2020-06-02T10:59:14.5884491Z (17/24) Installing libxml2 (2.9.10-r3)
2020-06-02T10:59:14.6101077Z (18/24) Installing libltdl (2.4.6-r7)
2020-06-02T10:59:14.6183570Z (19/24) Installing libxslt (1.1.34-r0)
2020-06-02T10:59:14.6303896Z (20/24) Installing xmlsec (1.2.29-r0)
2020-06-02T10:59:14.6472930Z (21/24) Installing oath-toolkit-libpskc (2.6.2-r0)
2020-06-02T10:59:14.6560934Z (22/24) Installing nettle (3.5.1-r0)
2020-06-02T10:59:14.6697173Z (23/24) Installing stoken (0.92-r1)
2020-06-02T10:59:14.9156453Z (24/24) Installing openconnect (8.10-r0)
2020-06-02T10:59:15.1798963Z Executing busybox-1.31.1-r9.trigger
2020-06-02T10:59:15.1862559Z OK: 255 MiB in 88 packages
```
After recent update stoken (0.92-r2) is used instead of stoken (0.92-r1)

As a workaround, I added nettle as an explicit dependency to Dockerfile, but received error anyway:
```
(28/31) Installing nettle (3.5.1-r0)
(29/31) Installing openssl-dev (1.1.1g-r0)
(30/31) Installing unixodbc (2.3.7-r2)
(31/31) Installing unixodbc-dev (2.3.7-r2)
Executing busybox-1.31.1-r9.trigger
OK: 207 MiB in 65 packages
fetch http://dl-3.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  so:libhogweed.so.6 (missing):
    required by: stoken-0.92-r2[so:libhogweed.so.6]
  so:libnettle.so.8 (missing):
    required by: stoken-0.92-r2[so:libnettle.so.8]
```

**Comment by Dongzx** [comment]
I encountered the same problem as above today.   When will this problem be resolved?

**Comment by oilegor1029** [comment]
I'm having the same problem.

**Comment by AndriyKarpyak** [comment]
As a workaround I switched to Debian but alpine is preferable.

**Comment by ncopa** [comment]
This happens because when you mix `v3.11` repos with `edge` and some library gets an ABI breaking update. In this case it was `nettle`.

To avoid this to happen you need only one repo, either `edge` or `v3.11` (or `3.12` which is out now).

I guess `stoken` could be moved to `community` repo now, since you apparently have ben using it in production for a while. Please file an issue on the [alpine linux gitlab](https://gitlab.alpinelinux.org/groups/alpine/-/issues) that you want move `stoken` to `community` and that you want back port it to `v3.12`. Add a link to this issue.

Thanks!

**Comment by AndriyKarpyak** [comment]
@ncopa Thank you for an explanation. I will create the request as You suggested and also play with repositories to resolve my problem.

As for me then I'm using it only for a few weeks, the project is only on the MVP stage.

**Comment by ncopa** [comment]
This is fixed in `alpine:3.12` now in commit alpinelinux/aports@e766baa626385f7b01147a3dc3fbb77666ca1309

You should no longer need `testing` repo for `stoken`.
